src/components/tips/tip-uploader.tsx

```diff
@@ -0,0 +1,301 @@
+'use client'
+import {z} from 'zod'
+import React from 'react'
+import cx from 'classnames'
+import toast from 'react-hot-toast'
+import {twMerge} from 'tailwind-merge'
+import {useRouter} from 'next/navigation'
+import VideoUploader from '@/components/upload/video-uploader'
+import {useForm} from 'react-hook-form'
+import {zodResolver} from '@hookform/resolvers/zod'
+import {trpc} from '@/app/_trpc/client'
+import Spinner from '@/components/spinner'
+import useFileUploadReducer from '@/hooks/use-file-upload-reducer'
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/ui/form'
+import {Textarea} from '@/ui/textarea'
+import {Input} from '@/ui/input'
+import {Button} from '@/ui/button'
+import slugify from 'slugify'
+
+function toChicagoTitleCase(slug: string): string {
+  const minorWords: Set<string> = new Set([
+    'and',
+    'but',
+    'for',
+    'or',
+    'nor',
+    'the',
+    'a',
+    'an',
+    'as',
+    'at',
+    'by',
+    'for',
+    'in',
+    'of',
+    'on',
+    'per',
+    'to',
+  ])
+
+  return slug
+    .replace(/-|_/g, ' ')
+    .split(' ')
+    .map((word, index, array) => {
+      if (
+        index > 0 &&
+        index < array.length - 1 &&
+        minorWords.has(word.toLowerCase())
+      ) {
+        return word.toLowerCase()
+      }
+      return word.charAt(0).toUpperCase() + word.substring(1).toLowerCase()
+    })
+    .join(' ')
+}
+
+function removeFileExtension(filename: string): string {
+  return filename.replace(/\.[^/.]+$/, '')
+}
+
+const formatString = (str: string) => {
+  return toChicagoTitleCase(removeFileExtension(str))
+}
+
+const NewTipFormSchema = z.object({
+  title: z.string().min(10).max(90),
+  description: z.string().min(2).max(240),
+  s3Url: z.string().min(4),
+})
+
+const TipCreationForm: React.FunctionComponent<
+  React.PropsWithChildren<any>
+> = () => {
+  const [s3Url, sets3Url] = React.useState<string>('')
+  const router = useRouter()
+  const form = useForm<z.infer<typeof NewTipFormSchema>>({
+    resolver: zodResolver(NewTipFormSchema),
+    defaultValues: {
+      title: '',
+      description: '',
```

Comment:
I plan on adding more tip metadata like tags and body but this gets it started

Reply:
I think the initial draft creation form can be very simple.